scene/3d/skeleton_3d.cpp

```diff
@@ -322,74 +376,86 @@ void Skeleton3D::_notification(int p_what) {
 				for (uint32_t i = 0; i < bind_count; i++) {
 					uint32_t bone_index = E->skin_bone_indices_ptrs[i];
 					ERR_CONTINUE(bone_index >= (uint32_t)len);
-					rs->skeleton_bone_set_transform(skeleton, i, bonesptr[bone_index].pose_global * skin->get_bind_pose(i));
+					rs->skeleton_bone_set_transform(skeleton, i, bonesptr[bone_index].global_pose * skin->get_bind_pose(i));
 				}
 			}
-			emit_signal(SceneStringNames::get_singleton()->pose_updated);
-		} break;
 
-#ifndef _3D_DISABLED
-		case NOTIFICATION_INTERNAL_PHYSICS_PROCESS: {
-			// This is active only if the skeleton animates the physical bones
-			// and the state of the bone is not active.
-			if (animate_physical_bones) {
-				for (int i = 0; i < bones.size(); i += 1) {
-					if (bones[i].physical_bone) {
-						if (bones[i].physical_bone->is_simulating_physics() == false) {
-							bones[i].physical_bone->reset_to_rest_position();
-						}
-					}
+			for (int i = 0; i < len; i++) {
+				bonesptr[i].modificating = false;
+			}
+			updating = false;
+			is_update_needed = false;
+		} break;
+		case NOTIFICATION_INTERNAL_PROCESS: {
+			if (modifier_callback_mode_process == MODIFIER_CALLBACK_MODE_PROCESS_IDLE) {
+				_find_modifiers();
+				if (!modifiers.is_empty()) {
+					_update_deferred();
 				}
 			}
 		} break;
-		case NOTIFICATION_READY: {
-			if (Engine::get_singleton()->is_editor_hint()) {
-				set_physics_process_internal(true);
+		case NOTIFICATION_INTERNAL_PHYSICS_PROCESS: {
```

Comment:
Should this not be called depending on using IDLE or physics? (not just physics like here)

Reply:
Ah right, I see if the process is registered correctly, there should be no need for if-blocking here.